static/js/zamboni/validator.js

```diff
@@ -384,9 +384,12 @@ function initValidator($doc) {
 
     CompatMsgVisitor.prototype.message = function(msg) {
         if (msg.for_appversions) {
-            if (this.hasMatchingAppVer(msg.for_appversions)) {
-                var app = {guid: guid, version: version, id: id};
-                msg.tier = id;  // change the tier to match app/version
+            var guid = this.findMatchingApp(msg.for_appversions)
+            if (guid) {
+                var app = {guid: guid, version: this.majorTargetVer[guid]};
+                app.id = (app.guid + '-' + app.version).replace(/[^a-z0-9_-]+/gi, '')
```

Comment:
Alas, I cannot. I accidentally deleted this line in the last commit, where I came across a comment saying to code did something completely different from what it actually did. I spent a couple of minutes trying to figure this out. Basically, it's some kind of filter string to decide if the message should be shown. But it doesn't make a lot of sense.